### ISSUE TITLE
Compiling error. (OpenWRT ar71xx MIPSBE)

### ISSUE THREAD

**Issue by adde88** [ZerBea/hcxtools#100]
Hi, looks like latest commits caused some issues when compiling for some systems.

I'm compiling for the WiFi Pineapples. (ar71xx CPU / MIPS-BE).
I noticed some updates lately regarding endianess, that's why i'm referencing what system i'm compiling for.

Here's the error message:
```
mips-openwrt-linux-uclibc-gcc -Os -pipe -mno-branch-likely -mips32r2 -mtune=34kc -fno-caller-saves -fhonour-copts -Wno-error=unused-but-set-variable -msoft-float -mips16 -minterlink-mips16  -MMD -MF .deps/hcxpcaptool.d -o hcxpcaptool hcxpcaptool.c -lz -lcrypto 
hcxpcaptool.c: In function 'processpcapng':
hcxpcaptool.c:5684:2: error: expected ';' before 'if'
  if(blocktype == PCAPNGBLOCKTYPE)
  ^
make[3]: *** [Makefile:79: hcxpcaptool] Error 1
```

**Comment by ZerBea** [comment]
Should be fixed by pull request from RealEnder:
https://github.com/ZerBea/hcxtools/commit/9c952fd701d741a552e8bc6f5a1ed52f03bd6a7b

Due to the results of some fuzzing tests, I decided to rewrite the pcapng engine completely. That caused this ugly issues and I hope they are fixed, now. Also I noticed an issue within pcapng option block handling of hcxdumptool and fixed it (now v 5.1.7).
Running hcxpcaptool against come some pcapng files captured on a mips system now showing the expected results.
Please test and reopen issue, if it doesn't work.

**Comment by adde88** [comment]
Will do, thanks for the awesome work on these tools :)
